packages/wallet/wdk/src/sequence/guard.ts

```diff
@@ -0,0 +1,72 @@
+import { Address, Hex, Signature, AbiParameters, Bytes, TypedData } from 'ox'
+import { Shared } from './manager.js'
+import { Kinds } from './types/signer.js'
+import { Payload, Signature as SequenceSignature } from '@0xsequence/wallet-primitives'
+
+export class Guard {
+  constructor(private readonly shared: Shared) {}
+
+  async sign(
+    wallet: Address.Address,
+    chainId: bigint,
+    payload: Payload.Payload,
+  ): Promise<SequenceSignature.SignatureOfSignerLeafHash> {
+    const digest = Payload.hash(wallet, chainId, payload)
+    const typedData = Payload.toTyped(wallet, chainId, payload)
+    const serialized = Hex.fromString(TypedData.serialize(typedData))
+
+    const auxData = AbiParameters.encode(AbiParameters.from(['address', 'uint256', 'bytes', 'bytes']), [
```

Comment:
It may not be the best time now but it would be nice to get rid of the concept of `auxData`, and instead just pass these arguments on the RPC request directly. It is something that no longer makes sense.